test/test_utils.py

```diff
@@ -29,6 +29,55 @@ def objective(label):
     assert time.time() - t0 < sleep_time / 2.0
 
 
+def test_cache_decorator_consistency():
+
+    cache_fn = tempfile.mkstemp()[1]
+
+    @gp.utils.disk_cache(cache_fn)
+    def objective_f(x):
+        return x
+
+    @gp.utils.disk_cache(cache_fn)
+    def objective_g(x):
+        return x ** 2
+
+    @gp.utils.disk_cache(cache_fn)
+    def objective_h(x):
+        return x
+
+    @gp.utils.disk_cache(cache_fn)
+    def objective_f_2d(x, y):
+        return x
+
+    @gp.utils.disk_cache(cache_fn)
+    def objective_f_with_kwargs(x, test=None):
+        return x
+
+    x = 2
+
+    assert objective_f(x) == pytest.approx(x)
+    assert objective_h(x) == pytest.approx(x)
+
+    # calling a different function decorated with same name but
```

Comment:
with "same name", you mean "same filename", right? Can you make this more clear then. Also for the other comments.

Reply:
good point, fixed